src/graph/graph.c

```diff
@@ -12,7 +12,32 @@
 #include "../GraphBLASExt/GxB_Delete.h"
 #include "../util/rmalloc.h"
 
-/*========================= Synchronization functions ========================= */
+static GrB_BinaryOp _graph_edge_accum = NULL;
+
+/* ========================= GraphBLAS functions ========================= */
+void _edge_accum(void *_z, const void *_x, const void *_y) {
+    EdgeID *z = (EdgeID*)_z;
+    const EdgeID *x = (const EdgeID*)_x;
+    const EdgeID *y = (const EdgeID*)_y;
+
+    EdgeID *ids;
+    /* Single edge ID,
+     * switching from single edge ID to multiple IDs. */
+    if(SINGLE_EDGE(*x)) {
+        ids = array_new(EdgeID, 2);
+        ids = array_append(ids, SINGLE_EDGE_ID(*x));
+        ids = array_append(ids, SINGLE_EDGE_ID(*y));
+        // TODO: Make sure MSB of ids isn't on.
```

Comment:
On my machines, it seems like the MSB is always 0 for pointers, but I have no idea how safely we can rely on that! And if it can be 1, can we safely modify it?